src/openapi3/handleJson.ts

```diff
@@ -11,68 +12,83 @@ export const removeTrailingSlash = (url: string): string => {
   return url;
 };
 
-export default function (
+/* Assumptions:
+  RESTful APIs typically have two paths per resources -  a `/noun` path and a 
+  `/noun/{id}` path. `getResources` strips out the latter, allowing us to focus
+  on the former.
+
+  In OpenAPI3, the `/noun` path will typically have a `get` action, that 
+  probably accepts parameters and would respond with a array of objects, 
+  described in the `items` field.
+*/
+
+export default async function (
   response: OpenAPIV3.Document,
   entrypointUrl: string
-): Resource[] {
-  const paths = getResources(response.paths);
+): Promise<Resource[]> {
+  const results: ResolvedRefsResults = await jsonRefs.resolveRefs(response);
+  const document: OpenAPIV3.Document = results.resolved as OpenAPIV3.Document;
+
+  const paths = getResources(document.paths);
+
+  let serverUrlOrRelative = "/";
+  if (document.servers) {
+    serverUrlOrRelative = document.servers[0].url;
+  }
+
+  const serverUrl = new URL(serverUrlOrRelative, entrypointUrl).href;
 
-  const resources = paths.map((item) => {
+  const resources: Resource[] = [];
+
+  paths.forEach((item) => {
```

Comment:
Why using a `forEach` and not a `map` like before?

Reply:
I conversations like this one: https://github.com/api-platform/api-doc-parser/issues/37
...there's a talk of being permissive about not trying to map every route to a resource. Open API 3 looks so different from a resource-based description that it seemed to make sense to just drop out for any path that doesn't match our expectations for a resource, rather than error.

Reply:
Makes sense, thanks for the explanation.